src/jit/lsraarmarch.cpp

```diff
@@ -403,13 +403,22 @@ void Lowering::TreeNodeInfoInitPutArgReg(
     info.srcCount++;
 
     // Set the register requirements for the node.
-    const regMaskTP argMask = genRegMask(argReg);
+    regMaskTP argMask = genRegMask(argReg);
+#ifdef ARM_SOFTFP
+    // If type of node is `long` then it is actually `double`.
+    // The actual `long` types must have been transformed as a field list with two fields.
+    if (node->TypeGet() == TYP_LONG)
+    {
+        info.srcCount++;
+        argMask |= genRegMask(REG_NEXT(argReg));
```

Comment:
This would be a good place to assert that `genRegArgNext(argReg == REG_NEXT(argReg)`, and note that the register allocator depends on the arg registers being in sequential order.